// This file was generated with the Protobuf generator tool.
// Dataset ID    : 40701
// Dataset Name  : churn;
// Dataset URL   : https://www.openml.org/data/v1/download/4965302/churn.arff
// Num. Columns  : 21
// Num. Rows     : 5000
// Target Feature: class

// Beginning of Description of Dataset:
// **Author**: Unknown  
// **Source**: [PMLB](https://github.com/EpistasisLab/penn-ml-benchmarks/tree/master/datasets/classification), [BigML](https://bigml.com/user/francisco/gallery/dataset/5163ad540c0b5e5b22000383), Supposedly from UCI but I can't find it there.  
// **Please cite**:   
// A dataset relating characteristics of telephony account features and usage and whether or not the customer churned. Originally used in [Discovering Knowledge in Data: An Introduction to Data Mining](http://secs.ac.in/wp-content/CSE_PORTAL/DataMining_Daniel.pdf).
//

syntax = "proto3";

message Empty {
}

message Features {
	uint32   State                          = 1;
	uint32   Account_length                 = 2;
	string   Area_code                      = 3;
	double   Phone_number                   = 4;
	string   International_plan             = 5;
	string   Voice_mail_plan                = 6;
	uint32   Number_vmail_messages          = 7;
	double   Total_day_minutes              = 8;
	uint32   Total_day_calls                = 9;
	double   Total_day_charge               = 10;
	double   Total_eve_minutes              = 11;
	uint32   Total_eve_calls                = 12;
	double   Total_eve_charge               = 13;
	double   Total_night_minutes            = 14;
	uint32   Total_night_calls              = 15;
	double   Total_night_charge             = 16;
	double   Total_intl_minutes             = 17;
	uint32   Total_intl_calls               = 18;
	double   Total_intl_charge              = 19;
	string   Number_customer_service_calls  = 20;
	string   Class                          = 21;
}

service get_next_row {
	 rpc get_next_row(Empty) returns(Features);
}

/*
//This code section can be directly used
//at the end of gen_next_row in the server.py file
//for OpenML dataset nr. 40701
        response.State                          = numpy.uint32(row[0])
        response.Account_length                 = numpy.uint32(row[1])
        response.Area_code                      = row[2]
        response.Phone_number                   = row[3]
        response.International_plan             = row[4]
        response.Voice_mail_plan                = row[5]
        response.Number_vmail_messages          = numpy.uint32(row[6])
        response.Total_day_minutes              = row[7]
        response.Total_day_calls                = numpy.uint32(row[8])
        response.Total_day_charge               = row[9]
        response.Total_eve_minutes              = row[10]
        response.Total_eve_calls                = numpy.uint32(row[11])
        response.Total_eve_charge               = row[12]
        response.Total_night_minutes            = row[13]
        response.Total_night_calls              = numpy.uint32(row[14])
        response.Total_night_charge             = row[15]
        response.Total_intl_minutes             = row[16]
        response.Total_intl_calls               = numpy.uint32(row[17])
        response.Total_intl_charge              = row[18]
        response.Number_customer_service_calls  = row[19]
        response.Class                          = row[20]
*/

